### PR TITLE
docs(component-store): add documentation for tapResponse operator

### LINKED ISSUE #2811: Document tapResponse
## Other information:

The `tapResponse` operator was introduced with #2763 but it's missing documentation.
I think it would be great if we had some example usage in our docs 🙂

## I would be willing to submit a PR for the docs :heart:

[ ] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

### PR DESCRIPTION
**Pull request by jordanpowell88**
Closes #2811

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[X] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #2811

## Does this PR introduce a breaking change?

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: projects/ngrx.io/content/guide/component-store/effect.md [hunk 1]
```diff
@@ -71,3 +71,21 @@ export class MovieComponent {
 
 }
 </code-example>
+
+## tapResponse
+An easy way to handle the response in ComponentStore effects in a safe way, without additional boilerplate is to use the `tapResponse` operator. It enforces that the error case is handled and that the effect would still be running should an error occur. It is essentially a simple wrapper around `tap` that handles success and error + `catchError(() => EMPTY)`.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/component-store/effect.md:76]: The `catchError(() => EMPTY)` is confusing me a little here,  is it missing something or is it just me 😅

Comment (alex-okrushko) [reply projects/ngrx.io/content/guide/component-store/effect.md:76]:
```suggestion
An easy way to handle the response in ComponentStore effects in a safe way, without additional boilerplate is to use the `tapResponse` operator. It enforces that the error case is handled and that the effect would still be running should an error occur. It is essentially a simple wrapper around two operators:
- `tap` that handles success and error
- `catchError(() => EMPTY)` that ensures that the effect continues to run after the error.
```

@timdeschryver Maybe something like this?

File: projects/ngrx.io/content/guide/component-store/effect.md [hunk 2]
```diff
@@ -71,3 +71,21 @@ export class MovieComponent {
 
 }
 </code-example>
+
+## tapResponse
+An easy way to handle the response in ComponentStore effects in a safe way, without additional boilerplate is to use the `tapResponse` operator. It enforces that the error case is handled and that the effect would still be running should an error occur. It is essentially a simple wrapper around `tap` that handles success and error + `catchError(() => EMPTY)`. 
+
+<code-example header="movies.store.ts">
+  readonly getMovie = this.effect((movieId$: Observable<string>) => {
```
Comment (alex-okrushko) [inline projects/ngrx.io/content/guide/component-store/effect.md:79]:
```suggestion
  readonly getMovie = this.effect((movieId$: Observable&lt;string&gt;) => {
```

**Review by alex-okrushko — CHANGES_REQUESTED** [review]
@jordanpowell88 
Thanks for the PR!

File: projects/ngrx.io/content/guide/component-store/effect.md (hunk 1 above)
Comment (timdeschryver) [reply projects/ngrx.io/content/guide/component-store/effect.md:76]: Yes, I think that's better. Thanks @alex-okrushko

**Comment by jordanpowell88** [comment]
> @jordanpowell88
> Thanks for the PR!

Your welcome! Its been a while 😢 but I now have a few hours a week that I can give to help out

**Review by alex-okrushko — APPROVED** [review]

**Review by timdeschryver — APPROVED** [review]

**Pull request merged**